### PR TITLE
shell: Improve setting and handling of os-release

### PR DESCRIPTION
**Pull request by petervo**
Avoid setting bad values in login.js but handle bad values better in shell

https://bugzilla.redhat.com/show_bug.cgi?id=1493756

### CODE REVIEW

File: pkg/shell/base_index.js [hunk 1]
```diff
@@ -757,7 +757,12 @@ var phantom_checkpoint = phantom_checkpoint || function () { };
 
         /* Branding */
         function setup_brand(id, default_title) {
-            var os_release = JSON.parse(window.localStorage['os-release'] || "{}");
+            var os_release = {};
+            try {
+                os_release = JSON.parse(window.localStorage['os-release'] || "{}");
+            } catch (ex) {
+                console.warn("Couldn't parse os-release", ex);
+            }
```
Comment (martinpitt) [inline pkg/shell/base_index.js]: I confirmed that `cockpit.format(s, dict)` just returns `s` with all macros replaced with an empty string for an empty `dict`, so this works reasonably well.

File: src/ws/login.js [hunk 2]
```diff
@@ -237,11 +237,13 @@ var phantom_checkpoint = phantom_checkpoint || function () { };
         if (authorized.indexOf("password") !== -1)
             id("authorized-input").checked = true;
 
-        var os_release = JSON.stringify(environment["os-release"]);
+        var os_release = environment["os-release"];
+        if (os_release)
+            window.localStorage.setItem('os-release', JSON.stringify(os_release));
```
Comment (martinpitt) [inline src/ws/login.js]: Does that actually make any difference? In JavaScript (unlike in Python), an empty dict is considered true. `environment["os-release"]` by itself is always set in cockpithandlers.c `build_environment()` AFAICS.

Comment (petervo) [reply src/ws/login.js]: Yes this makes a difference. Old code put the string "undefined" in localStorage.

Comment (martinpitt) [reply src/ws/login.js]: Oh, so it will be `undefined`, not an empty array, I see. I didn't see how it got from the above `base_index.js` to here.

**Pull request merged**